### ISSUE TITLE
[CAL-3029] 2024-05-23 - App store install flow

### ISSUE THREAD

**Issue by sean-brydon** [calcom/cal.com#13414]
Take over from @morgans existing work and complete the following tasks

refactor EventTypeAppInterfaces of the apps used in event-types settings

- [X] split Form part from the APPcard part
- [X] already done for qr_code app ([commit here](https://github.com/calcom/cal.com/pull/11975/commits/ca0edf1ee7165b74195eb1da24ddbf408a84e12e))
- [X] check if there are new oAuth apps added, if yes add isOAuth to app_metadata.ts
- [X] adapt oauth apps api callback / add to redirect to new app install flow
- [X] When clicking on install new app button, only redirect to the new app install flow if it is an oAuth app or an app that extends the EventType feature in it's config
- [X] pr was done commit by commit for easier readability, also self review comments on the github pr

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3029](https://linear.app/calcom/issue/CAL-3029/app-store-install-flow)</sub>

**Comment by keithwillcode** [comment]
Putting this back into review so we do another pass of tests on it before releasing. @joeauyeung @somay_chauhan 
@ebruinsma is currently pulling this commit out of main to unblock today's release

**Comment by CarinaWolli** [comment]
@somay_chauhan can this be closed or do we still need tests here?

**Comment by joeauyeung** [comment]
@CarinaWolli still looking for a second review here as it's labelled as high risk: https://github.com/calcom/cal.com/pull/14616

**Comment by keithwillcode** [comment]
Pulling back to In Progress since there are sub-issues outstanding. Main PR is merged though. Great stuff!
